lib/include/iot_taskpool.h

```diff
@@ -481,11 +487,26 @@ IotTaskPoolError_t IotTaskPool_GetStatus( IotTaskPool_t * const pTaskPool,
  *
  */
 /* @[declare_taskpool_trycancel] */
-IotTaskPoolError_t IotTaskPool_TryCancel( IotTaskPool_t * const pTaskPool,
-                                          IotTaskPoolJob_t * const pJob,
+IotTaskPoolError_t IotTaskPool_TryCancel( IotTaskPool_t const pTaskPool,
+                                          IotTaskPoolJob_t const pJob,
                                           IotTaskPoolJobStatus_t * const pStatus );
 /* @[declare_taskpool_trycancel] */
 
+/**
+ * @brief Returns a pointer to the job storage from an instance of a job handle 
+ * of type @ref IotTaskPoolJob_t. This function is guarateed to succeed for a 
+ * valid job handle.
+ *
+ * @param[in] pJob The job handle.
+ *
+ * @return A pointer to the storage associated with the job handle `pJob`.
+ * 
+ * @warning If the `pJob` handle used is invalid, the results will be undefined.
+ */
+/* @[declare_taskpool_getjobstoragefromhandle] */
+IotTaskPoolJobStorage_t * IotTaskPool_GetJobStorageFromHandle( IotTaskPoolJob_t pJob );
```

Comment:
Does this need to be part of the public API?  I can't think of a case where the application would call this.

Reply:
Anytime the app may need to retrieve the storage, e.g. to de-allocate it.